lib/robots/viomi/ViomiMapParser.js

```diff
@@ -458,6 +488,11 @@ class ViomiMapParser {
             //Repeated for every room there is
             this.take(unkLength * 2, "room data");
 
+            if (this.mapId === 0) {
+                // Sometimes the map is truncated here, when that happens the second word is zeroes
```

Comment:
Could these be incremental map updates or do those still contain a full map image etc?

Reply:
Nope, it contains everything except the "extended" room data + room outlines.

For some reason this is sent as soon as the robot touches the dock and in no other circumstance.

Reply:
Maybe thats when room splitting stuff is recalculated or something like that 🤔

Reply:
Indeed I just noticed it wants the same map ID for editing segments. It probably returns map ID set to 0 so that segment editing won't work until it's not done.